src/state_machine.zig

```diff
@@ -188,18 +190,20 @@ pub fn StateMachineType(comptime Storage: type, comptime constants_: struct {
             self: *StateMachine,
             comptime operation: Operation,
             input: []u8,
-        ) void {
-            var sum_reserved_timestamps: usize = 0;
+        ) !void {
+            var input_timestamp_or: usize = 0;
```

Comment:
This should be a `u64`. Unfortunately, unlike Rust, Zig won't complain about a mismatched `usize` vs `u64` like this unless you actually build for 32-bit.

Reply:
excellent catch, thanks!